### PR TITLE
docs: add TextEditor ADR

### PR DESCRIPTION
**Pull request by connorhaugh**
This ADR proposes a Paragon TextEditor component as a pass through of Tinymce, an existing WYSIWIG text editor, with standardization. This follows the suggested reverse process for engineers to submit things to pwg

### CODE REVIEW

File: docs/decisions/0017-Text-Editor-Using-TinyMce [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+Decision
+------
+Paragon will expose a TextEditor Component, which uses TinyMCE’s React library as a foundation. The TextEditor component will have a standardized list of plugins that will also be supported by Paragon, and will not need to be managed by the consumer. A default experience will also be provided for out-of-the-box usage. Any plugins beyond those already managed by paragon, paid or otherwise, will be up to the consumer to maintain. Additional extensions, like custom plugins for specific use cases, will live elsewhere, and not within Paragon. 
+Status
+------
+Proposed
+Context
+-------
+Already, there are two simultaneous ongoing implementations of React-based html editors using tinyMCE. One is the Text Xblock editor, the other is the instructor bulk email tool found in the communications MFE. Already, edX also uses tinyMCE in edx-platform, and hosts its own image of the service there. Presumably, this experience too will be moved from the ball of mud into an independent MFE, and other use cases for text editing will arise. Therefore, it seems obvious to standardize across implementations.
```
Comment (xitij2000) [inline docs/decisions/0017-Text-Editor-Using-TinyMce:14]: Make that three :-) The discussions MFE also uses it for post editing: https://github.com/openedx/frontend-app-discussions/blob/master/src/components/TinyMCEEditor.jsx

File: docs/decisions/0017-Text-Editor-Using-TinyMce [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Status
+------
+Proposed
+Context
+-------
+Already, there are Three simultaneous ongoing implementations of React-based html editors using tinyMCE. One is the Text Xblock editor, the other is the instructor bulk email tool found in the communications MFE. The final is the discussions MFE. Already, edX also uses tinyMCE in edx-platform, and hosts its own image of the service there. Presumably, this experience too will be moved from the ball of mud into an independent MFE, and other use cases for text editing will arise. Therefore, it seems obvious to standardize across implementations. Paragon will serve as the source of truth for TinyMce versioning, much in the same way the Data table component works.
```
Comment (adamstankiewicz) [inline docs/decisions/0017-Text-Editor-Using-TinyMce:14]: One nit might be consistency with how TinyMCE is spelled throughout :)

File: docs/decisions/0017-Text-Editor-Using-TinyMce [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+TinyMCE itself is “based on WCAG [2] and Section 508, to ensure that the TinyMCE editor is accessible to content authors.” Other alternatives do not provide this assurance. In addition to being a familiar entity, it also accomplishes our main goals for content authoring.
```
Comment (adamstankiewicz) [inline docs/decisions/0017-Text-Editor-Using-TinyMce:15]: Is there a link that would be handy to include here?

**Review by adamstankiewicz — APPROVED** [review]
LGTM with a couple nits. Feel free to change the status to "Approved" before you merge.

**Comment by muselesscreator** [comment]
I believe this makes sense in the long term, but for the short-term, the team needs w/rt the tinyMCE installation are going to leave them with a much less configurable option than we would want for Paragon consumption, and the TnL squad will not have the bandwidth to continue pressing on this from a component standardization perspective, so that work will need to get forwarded to the FWG or PWG for future ownership.

**Comment by adamstankiewicz** [comment]
Following up on the previous offline discussions in Frontend/Paragon Working Groups, we decided we'd defer on considering a `TextEditor` component in Paragon, at least one that consumes TinyMCE (especially needing to support paid plugins).

That said, it may be interesting if Paragon one day supported its own custom text editor component that didn't rely on TinyMCE, but instead something like https://github.com/facebook/lexical. 

As a result, I'm inclined to close the PR for now, though we'll definitely be considering whether a text editor makes sense on the long term roadmap for Paragon. 

@connorhaugh If you disagree, please feel to re-open the issue 😄 Thanks!

**Pull request closed without merging**